examples/plot_classification_example.py

```diff
@@ -0,0 +1,244 @@
+"""
+Measuring Individual and Group Variable Importance for Classification
+======================================================================
+In this example, we show on the Iris dataset how to measure variable importance for
+classification tasks. We use three different variable importance methods: CPI, LOCO and
+PermutationImportance with two different classifiers: LogisticRegressionCV and
+LinearSVC. We start by measuring the importance of individual variables and then show
+how to measure the importance of groups of variables.
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from joblib import Parallel, delayed
+from scipy.stats import ttest_1samp
+from sklearn.base import clone
+from sklearn.datasets import load_iris
+from sklearn.linear_model import LogisticRegressionCV, RidgeCV
+from sklearn.metrics import balanced_accuracy_score, hinge_loss, log_loss
+from sklearn.model_selection import GridSearchCV, KFold
+from sklearn.svm import SVC
+
+from hidimstat import CPI, LOCO, PermutationImportance
+
+########################################################################
+# Load the iris dataset and add a spurious feature
+# ----------------------------------------------------------------------
+# We load the iris dataset and add a spurious feature that is a linear combination of
+# the petal length, width amd some noise but not related to the target. The spurious feature
+# allows to illustrate that `PermutationImportance` is not robust to spurious features,
+# contrarily to `LOCO`` and `CPI``.
+dataset = load_iris()
+rng = np.random.RandomState(0)
+X, y = dataset.data, dataset.target
+spurious_feat = X[:, 2] + X[:, 3]
+spurious_feat += rng.normal(size=X.shape[0], scale=np.std(spurious_feat) / 2)
+X = np.hstack([X, spurious_feat.reshape(-1, 1)])
+
+dataset.feature_names = dataset.feature_names + ["spurious_feat"]
+
+
+############################################################################
+# Measure variable importance
+# --------------------------------------------------------------------------
+# Since all three methods compute variable importance as a loss difference, they
+# require a K-fold cross-fitting. Computing the importance for each fold is
+# embarassingly parallel, so we use joblib to parallelize the computation.
+def run_one_fold(X, y, model, train_index, test_index, vim_name="CPI", groups=None):
+    model_c = clone(model)
+    model_c.fit(X[train_index], y[train_index])
+    y_pred = model_c.predict(X[test_index])
+
+    if isinstance(model_c, LogisticRegressionCV):
+        method = "predict_proba"
+        loss = log_loss
+        model_name = "LogReg"
+    else:
+        method = "decision_function"
+        loss = hinge_loss
+        model_name = "SVC"
+
+    if vim_name == "CPI":
```

Comment:
It's better, for example, to put the creation of the object outside of the function and add a short description of the methods. Most people don't know CPI, LOCO and PFI.

Reply:
as discussed below, it has to be in the loop since it requires the fitted model

Reply:
see my response below.